Cython/Compiler/ModuleNode.py

```diff
@@ -1057,30 +1057,17 @@ def generate_cpp_class_definition(self, entry, code):
                 if attr.type.is_cfunction and attr.type.is_static_method:
                     code.put("static ")
                 elif attr.name == "<init>":
-                    constructor = attr
+                    #constructor = attr
+                    constructor = scope.lookup_here("<init>")
                 elif attr.name == "<del>":
                     destructor = attr
                 elif attr.type.is_cfunction:
                     code.put("virtual ")
                     has_virtual_methods = True
                 code.putln("%s;" % attr.type.declaration_code(attr.cname))
             is_implementing = 'init_module' in code.globalstate.parts
-            if constructor or py_attrs:
-                if constructor:
-                    arg_decls = []
-                    arg_names = []
-                    for arg in constructor.type.original_args[
-                            :len(constructor.type.args)-constructor.type.optional_arg_count]:
-                        arg_decls.append(arg.declaration_code())
-                        arg_names.append(arg.cname)
-                    if constructor.type.optional_arg_count:
-                        arg_decls.append(constructor.type.op_arg_struct.declaration_code(Naming.optional_args_cname))
-                        arg_names.append(Naming.optional_args_cname)
-                    if not arg_decls:
-                        arg_decls = ["void"]
-                else:
-                    arg_decls = ["void"]
-                    arg_names = []
+
+            def generate_cpp_constructor_code(arg_decls, arg_names, is_implementing, py_attrs, constructor):
```

Comment:
Yep, you're right. Given that `generate_cpp_constructor_code` is a helper function used only in `generate_cpp_class_definition`, it would be sensible to put it at the same scope, right ? That would be class scope then.